### DISCUSSION TITLE
How do I access the main/function specific flowgraph from the Python API?

### DISCUSSION THREAD

**Discussion by houseofwealth** [Vector35/binaryninja-api#2826] [API] [answered]
The title pretty much says it all. I have a exe I'm analyzing in Binary Ninja and I want to access (and ultimately modify) the displayed flowgraph edges (such as color) and add some annotations to the nodes. I went through the API documentation and I was unable to see this anywhere. Somewhat related to this, what's the difference between a CoreFlowGraph and a FlowGraph?

thanks

**Comment by fuzyll** [comment, marked as answer]
Hey, sorry it took so long to get an answer to this.

There's two answers to your question:

1. If you want to make a change to an existing FlowGraph, you can use the highlight API (https://api.binary.ninja/binaryninja.highlight-module.html?highlight=highlight#module-binaryninja.highlight). But, you cannot change edges using that API on an existing graph.
2. You can fabricate your own graph (as a new window or view, however you want to do it) like this snippet does: https://gist.github.com/psifertex/6fbc7532f536775194edd26290892ef7#file-flowgraph-py. This gives you full access to customize edges, texture, color, etc.

Generally speaking, if "Core" is in the name, it's probably not intended for public use. Those are usually used for internal implementation details.
